docs/dpdk/gnmi-ctl.rst

```diff
@@ -0,0 +1,288 @@
+gnmi-ctl executable
+-------------------
+
+gnmi-ctl is a gRPC-based C++ network management interface client to handle port configurations and program fixed functions in the P4 pipeline. This client connects
+to the Config monitoring service running as part of Stratum and integrated as part of
+infrap4d. This utility is built as part of networking recipe and is available under
+`install/bin/`.
+
+gnmi client connects to gRPC port 9339 opened by the server and sends a protobuf
+based message. Refer to gnmi.proto and gnmi_ext.proto for more details on type of
+messages and services that are available for a gNMI client.
+
+Each CLI format should be inline with the YANG parser available in server which
+is a tree based hierarchy. Tree has `interfaces` as root-node, followed by the
+device type. Each device can have multiple ports, where port-name acts as a KEY.
+SET command can be executed to configure the ports and GET command to retrieve the
+information about previously configured ports
+
+1) Set attributes for a vhost port::
```

Comment:
One colon, not two. This comment applies throughout the file.

Reply:
Double colon is required for rst formatting

Reply:
Ah, so desu ka.